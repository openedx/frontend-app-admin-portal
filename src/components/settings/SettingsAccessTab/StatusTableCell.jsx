import React from 'react';
import PropTypes from 'prop-types';
import { Badge } from '@edx/paragon';

const StatusTableCell = ({ row }) => {
  const { isValid } = row.original;
  return (
    <Badge variant={isValid ? 'success' : 'light'}>
      {isValid ? 'Active' : 'Inactive'}
    </Badge>
  );
};

StatusTableCell.propTypes = {
  row: PropTypes.shape({
    original: PropTypes.shape({
      isValid: PropTypes.bool,
    }),
  }).isRequired,
};

export default StatusTableCell;
