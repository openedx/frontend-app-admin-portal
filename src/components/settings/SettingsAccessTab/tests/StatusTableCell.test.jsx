import React from 'react';
import renderer from 'react-test-renderer';

import StatusTableCell from '../StatusTableCell';

describe('StatusTableCell', () => {
  it('renders valid status correctly', () => {
    const props = {
      row: {
        original: {
          isValid: true,
        },
      },
    };
    const tree = renderer
      .create(<StatusTableCell {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders invalid status correctly', () => {
    const props = {
      row: {
        original: {
          isValid: false,
        },
      },
    };
    const tree = renderer
      .create(<StatusTableCell {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
