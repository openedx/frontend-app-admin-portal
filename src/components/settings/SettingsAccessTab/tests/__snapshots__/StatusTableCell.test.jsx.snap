// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusTableCell renders invalid status correctly 1`] = `
<span
  className="badge badge-light"
>
  Inactive
</span>
`;

exports[`StatusTableCell renders valid status correctly 1`] = `
<span
  className="badge badge-success"
>
  Active
</span>
`;
